lib/galaxy/datatypes/converters/archive_to_directory.xml

```diff
@@ -0,0 +1,37 @@
+<tool id="CONVERTER_archive_to_directory" name="Unpack archive to directory" version="1.0.0" profile="21.09">
+    <!-- Use compression_utils instead of shell commands (tar/unzip) so we can verify safety of results -->
+    <requirements>
+        <requirement type="package" version="23.2.1">galaxy-util</requirement>
+    </requirements>
+    <command><![CDATA[
+        mkdir '$output1.files_path' &&
+        cd '$output1.files_path' &&
+        python -c "from galaxy.util.compression_utils import CompressedFile; CompressedFile('$input1').extract('.')"
+    ]]></command>
+    <configfiles>
+        <configfile filename="metadata_json"><![CDATA[{
+    "output1": {
+        "name": "$input1.name unpacked to $__target_datatype__",
+        "ext": "$__target_datatype__"
+    }
+}]]></configfile>
```

Comment:
Hmm, somehow this doesn't work anymore...

When I run the converter to `directory` the datatype changes to `directory`, but I don't get the updated name or the blurb with the total size set in peek. I'm not sure what I'm missing here... :thinking: 

![image](https://github.com/user-attachments/assets/a47a2b2b-2b52-4b0b-afaa-f57963776447)

Reply:
It looks like I was missing `cp '$metadata_json' galaxy.json` in the command, but now I get a template error:

```
Traceback (most recent call last):
  File "/home/dlopez/dev/galaxy/lib/galaxy/tools/evaluation.py", line 107, in global_tool_logs
    return func()
           ^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/tools/evaluation.py", line 634, in _build_command_line
    command_line = fill_template(
                   ^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/template.py", line 115, in fill_template
    return fill_template(
           ^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/template.py", line 115, in fill_template
    return fill_template(
           ^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/template.py", line 115, in fill_template
    return fill_template(
           ^^^^^^^^^^^^^^
  [Previous line repeated 7 more times]
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/template.py", line 123, in fill_template
    raise first_exception or e
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/template.py", line 87, in fill_template
    return unicodify(t, log_exception=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/util/__init__.py", line 1182, in unicodify
    value = str(value)
            ^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/.venv/lib/python3.11/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "cheetah_DynamicallyCompiledCheetahTemplate_1728642092_1426992_84394.py", line 87, in respond
NameMapper.NotFound: cannot find 'metadata_json'
```

Reply:
It's just `cp 'metadata_json' galaxy.json` if you're using the `filename` argument, the previous version would only work with `<configfile name="metadata_json">`

Reply:
Thank you! That fixed the template error, but the metadata is still not being processed for some reason...

There is also an error when trying to cleanup the job directory:

```
galaxy.jobs ERROR 2024-10-11 12:49:34,434 [pN:main.1,p:229474,tN:LocalRunner.work_thread-0] Unable to cleanup job 11652
Traceback (most recent call last):
  File "/home/dlopez/dev/galaxy/lib/galaxy/jobs/__init__.py", line 2160, in cleanup
    self.object_store.delete(self.get_job(), base_dir="job_work", entire_dir=True, obj_dir=True)
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 580, in delete
    return self._invoke(
           ^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 485, in _invoke
    return self.__getattribute__(f"_{delegate}")(obj=obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1202, in _delete
    return self._call_method("_delete", obj, False, False, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1524, in _call_method
    return self._resolve_backend(object_store_id).__getattribute__(method)(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1068, in _delete
    path = self._get_filename(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1114, in _get_filename
    raise ObjectNotFound
galaxy.exceptions.ObjectNotFound: No such object found.
```

I guess it is expecting a "main" file for the dataset?

Reply:
That's an attempt to delete the job working directory, I don't think this is related.

Reply:
`dataset.blurb = "done"` ... this is where the done comes from. I would check out then why the datatype-specific metadata isn't being set. That said ... why use the galaxy.json method at all if you know the output format up front ?

Reply:
Yeah sorry I wasn't clear. The metadata not being set is, of course, not related at all to this other error attempting to cleanup the job directory. I thought it might have something to do with the `directory` datatype itself and worth mentioning.

Going back to the issue with the converter output metadata:

I thought that using `<configfile filename="metadata_json">` and then `<outputs provided_metadata_file="metadata_json">` will directly work (without having to `cp` any file). But I must have misunderstood something because it didn't work.

Curiously enough, the only way I got the metadata to work is by trying to `cp metadata_json metadata_json`, which doesn't make sense to me and obviously gives a command error but is the only way I got the output name changed...

![image](https://github.com/user-attachments/assets/851bb8f1-6414-4715-8104-b2763bf75537)

So definitely I'm missing something silly.

Reply:
> why use the galaxy.json method at all if you know the output format up front ?

That is a good question. I'm trying to continue what was already done so I don't know the exact reason. I guess it is to set the output name to "X unpacked to Y", and the target format. I assume right now the format will be `directory` but when we have the `zarr` datatype it should be able to set it to that as well so maybe we can't assume it is always `directory`.

Is there a better way?

Reply:
You can just use a normal label attribute on the output to do that.

> I thought that using `<configfile filename="metadata_json">` and then `<outputs provided_metadata_file="metadata_json">` will directly work (without having to `cp` any file). But I must have misunderstood something because it didn't work.

That is the idea, see also https://github.com/galaxyproject/galaxy/blob/dev/test/functional/tools/tool_provided_metadata_9.xml#L20-L22

Reply:
> That is the idea, see also https://github.com/galaxyproject/galaxy/blob/dev/test/functional/tools/tool_provided_metadata_9.xml#L20-L22

I saw that example, and I did "exactly" the same thing (even renamed variables to match it), but unfortunately, it didn't work, so there must be something slightly different going on... I'll keep investigating...

Reply:
Turns out the converter is perfectly fine, the issue was caused by setting `dataset.blurb = nice_size(dataset.dataset.total_size)`  in https://github.com/galaxyproject/galaxy/pull/17614/files#diff-38a85eadce6179cb91c77072b0617e6a8f652869c5b3fb1d76740909c3f24027R1222 if I replace the blurb with something else like "Directory of files" everything works... 

![image](https://github.com/user-attachments/assets/a4f205b8-b33c-4711-9137-558b199e1210)


So I understand this is not the correct way of getting the total uncompressed size?

Reply:
I mean this is kind of a useless blurb considering we store the total size elsewhere. There's a try/except that might hide the real exception.

Reply:
I see... the "done" blurb was extremely confusing, indeed it shows when there is an exception in the set_peek function, I'm not sure if that is a bug or by design.

If this blurb is not useful, then I will just drop the entire `set_peek`. Thanks!